consensus/blake3pow/consensus.go

```diff
@@ -635,7 +635,16 @@ func (blake3pow *Blake3pow) Finalize(chain consensus.ChainHeaderReader, header *
 			blake3pow.logger.Info("Indexed genesis utxos")
 		}
 	}
-	header.Header().SetUTXORoot(state.UTXORoot())
+	if err := state.SetUTXOSetSizes(); err != nil {
+		blake3pow.logger.WithError(err).Error("Failed to set UTXO set sizes")
```

Comment:
this will cause the block to get rejected, but this error isn't the blocks fault. This error is a failure in the node's state. This could cause miners to disagree/fork on blocks, just because their own state became inconsistent.

Instead, this should be a fatal error which crashes the node. Same for the error below this, and other similar errors.